lightning/src/ln/functional_test_utils.rs

```diff
@@ -2768,8 +2768,12 @@ pub fn do_pass_along_path<'a, 'b, 'c>(args: PassAlongPathArgs) -> Option<Event>
 							},
 						}
 						assert_eq!(*amount_msat, recv_value);
-						assert!(node.node.list_channels().iter().any(|details| details.channel_id == via_channel_id.unwrap()));
-						assert!(node.node.list_channels().iter().any(|details| details.user_channel_id == via_user_channel_id.unwrap()));
+						for (chan_id, user_chan_id) in inbound_channel_ids {
+							assert!(node.node.list_channels().iter().any(|details| &details.channel_id == chan_id));
+							if let Some(user_id) = user_chan_id {
```

Comment:
Could probably first `node.node.list_channels().iter().find(|details| &details.channel_id == chan_id)` and then see of user_id matches too?

Reply:
Makes sense! Updated in [pr3655.13](https://github.com/shaavan/rust-lightning/commits/pr3655.13)
Thanks!